src/item/abstract_item.cpp

```diff
@@ -212,7 +212,7 @@ void AbstractItem::contextMenuEvent(QGraphicsSceneContextMenuEvent* event)
     ungrabMouse(); //Important line! Otherwise the element will be moved after you close the context menu and draw a selection rectangle somewhere.
 
     if (actionSel == actionRemove) {
-        remove();
+        disconnectConnections();
         delete this;
```

Comment:
We should set `this = nullptr` after this to prevent accessing it after the memory has been freed.